gateway/controller.go

```diff
@@ -165,3 +169,65 @@ func (c *ControllerHTTPClient) GetAssignment(jobNamespace, instance string) (ass
 	}
 	return
 }
+
+// SetNamespace sets the namespace which the ControllerHTTPClient connects to
+func (c *ControllerHTTPClient) SetNamespace(namespace string) {
+	c.namespace = namespace
+}
+
+// FetchAllSchemas fetches all schemas
+func (c *ControllerHTTPClient) FetchAllSchemas() (tables []common.Table, err error) {
+	return  c.GetAllSchema(c.namespace)
+}
+
+// FetchSchema fetch one schema for given table
+func (c *ControllerHTTPClient) FetchSchema(tableName string) (table *common.Table, err error) {
+	request, err := c.buildRequest(http.MethodGet, fmt.Sprintf("schema/%s/tables/%s", c.namespace, tableName), nil)
+	if err != nil {
+		return
+	}
+	table = &common.Table{}
+	err = c.getJSONResponse(request, table)
+	if err != nil {
+		err = utils.StackError(err, "controller client error fetching schema for table: %s", tableName)
+		return
+	}
+	return
+}
+
+// FetchAllEnums fetches all enums for given table and column
+func (c *ControllerHTTPClient) FetchAllEnums(tableName string, columnName string) (enumDictReponse []string, err error) {
+	request, err := c.buildRequest(http.MethodGet, fmt.Sprintf("enum/%s/%s/columns/%s/enum-cases", c.namespace, tableName, columnName), nil)
+	if err != nil {
+		return
+	}
+
+	err = c.getJSONResponse(request, &enumDictReponse)
+	if err != nil {
+		err = utils.StackError(err, "controller client error fetching schema for table: %s", tableName)
+		return
+	}
+	return
+}
+
+// ExtendEnumCases extends enum cases to given table column
+func (c *ControllerHTTPClient) ExtendEnumCases(tableName, columnName string, enumCases []string) (enumIDs []int, err error) {
+	enumCasesRequest := client.EnumCasesWrapper{
```

Comment:
this seems like an unnecessary dependency on client package

Reply:
removed, done